tools/clang/lib/SPIRV/EmitSPIRVAction.cpp

```diff
@@ -673,12 +747,36 @@ class SPIRVEmitter : public ASTConsumer {
       const uint32_t resultType = typeTranslator.translateType(toType);
       return theBuilder.createLoad(resultType, fromValue);
     }
+
+    case CastKind::CK_HLSLVectorSplat: {
+      const size_t size = hlsl::GetHLSLVecSize(expr->getType());
+      const uint32_t scalarValue = doExpr(subExpr);
+
+      // Just return the scalar value for vector splat with size 1
+      if (size == 1) {
+        return scalarValue;
+      }
+
+      const uint32_t vecTypeId = typeTranslator.translateType(toType);
+      llvm::SmallVector<uint32_t, 4> elements(size, scalarValue);
+
+      return theBuilder.createCompositeConstruct(vecTypeId, elements);
+    }
+    case CastKind::CK_HLSLVectorToScalarCast: {
+      // We should have already treated vectors of size 1 as scalars.
+      // So do nothing here.
+      if (hlsl::GetHLSLVecSize(subExpr->getType()) == 1) {
+        return doExpr(subExpr);
+      }
+      emitError("vector to scalar cast unimplemented");
```

Comment:
I'm wondering: Is there any valid case of casting a vector larger than 1 element to a scalar?